rankers/BiMatchRanker/__init__.py

```diff
@@ -52,17 +51,18 @@ class BiMatchRanker(Chunk2DocRanker):
 
     So for every `Document`, a score is computed adding some penalty to the `missing` in both perspectives.
 
+    :param query_required_keys: requested fields for query
+    :param match_required_keys: requested fields for match
     :param d_miss: Cost associated to a miss chunk
     :param args:  Additional positional arguments
     :param kwargs: Additional keyword arguments
 
     .. warning:: Here we suppose that the smaller chunk score means the more similar.
     """
-    query_required_keys = ('length', )
-    match_required_keys = ('length', )
 
-    def __init__(self, d_miss: Optional[Union[int, float]] = None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, query_required_keys: Optional = ('length',), match_required_keys: Optional = ('length',),
```

Comment:
I think these should not be parameters, but they should be passed directly to the `super`constructor.

Reply:
In that case, the following one may be better? It depends on if we want to expose these params to users and the intention you refactored before,
`query_required_keys = ('length', )
match_required_keys = ('length', )

def init(self, d_miss: Optional[Union[int, float]] = None, *args, **kwargs):
super().init(self.query_required_keys, self. match_required_keys, *args, **kwargs)`

Reply:
For now in this case, I think we do not want to expose it to users.

Reply:
Sure, I will change to the second one. Or any other suggestions? 😄

Reply:
The second one, and passed by value if possible I would say

Reply:
Sure. Direct pass values now